src/reanimated2/animation/decay/decay.ts

```diff
@@ -1,36 +1,34 @@
 'use strict';
 import { defineAnimation, getReduceMotionForAnimation } from '../util';
 import type {
-  Animation,
   AnimationCallback,
   Timestamp,
-  ReduceMotion,
+  RequiredKeys,
+  Animation,
 } from '../../commonTypes';
+import { rubberBandEffectDecay } from './rubberBandDecay';
 import type {
   DecayAnimation,
+  DecayConfig,
   DefaultDecayConfig,
   InnerDecayAnimation,
 } from './utils';
-import { rubberBandDecay } from './rubberBandDecay';
+import { validateConfig } from './utils';
 import { rigidDecay } from './rigidDecay';
 
-interface DecayConfig {
-  deceleration?: number;
-  velocityFactor?: number;
-  rubberBandEffect?: boolean;
-  clamp?: number[];
-  velocity?: number;
-  reduceMotion?: ReduceMotion;
-}
-
-export type WithDecayConfig = DecayConfig;
-
 // TODO TYPESCRIPT This is a temporary type to get rid of .d.ts file.
 type withDecayType = (
   userConfig: DecayConfig,
   callback?: AnimationCallback
 ) => number;
 
+function isDefaultDecayConfigWithClamp(
+  config: DefaultDecayConfig
+): config is RequiredKeys<DefaultDecayConfig, 'clamp'> {
```

Comment:
If we have a separate type we must define it inside `utils.ts` to avoid circular dependencies 🔁, so `isDefaultDecayConfigWithClamp` should be in utils too. Because some 🧑‍💻  in the future could add more properties to the type and forget about type-checking function.